DEVDOCS-4620: [new] Data Layer API, add new spec (#1137)
bigcommerce/api-specs: docs/data-layer-api.md

# Data Layer API Reference

The [Data Layer API](/api-docs/store-management/data-layer-api) is a GraphQL management API that allows you to enable and disable the [Big Open Data Layer](/api-docs/analytics/bodl-for-storefronts) for a store.

**Request URL:** 

> `https://api.bigcommerce.com/stores/{store_hash}/v3/...`

Author: @solofeed Do you have the request url?

Reviewer: This is the graphql endpoint
`stores/{store_hash}/graphql`

**Path Parameters:**

| Name | Type | Description | Required |
| :--- | :--- | :--- | :--- |
| `store_hash` | string | Unique ID for a store | Yes |

## Authentication

### OAuth scopes

| UI Name | Permission | Parameter |
| :--- | :--- | :--- |

> |  | modify |  |

Author: @solofeed I assume the GraphQL DataLayer API authenticates using Store API Accounts (like the management APIs that are REST APIs)? Do you have the UI name and parameter for the OAuth Scopes?

Reviewer: scope - Information
permission - Manage Web Analytics

</tr>
</thead>
<tbody>
<tr>
<td colspan="2" valign="top"><strong>settings</strong></td>
<td valign="top"><a href="#storesettingsmutations">StoreSettingsMutations</a></td>
<td>

Store Settings mutations

</td>
</tr>
</tbody>
</table>

### Objects

#### DataSolutionsMutations

Data solutions mutations.

<table>
<thead>
<tr>
<th align="left">Field</th>
<th align="right">Argument</th>
<th align="left">Type</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td colspan="2" valign="top"><strong>updateDataLayer</strong></td>
<td valign="top"><a href="#updatedatalayerresult">UpdateDataLayerResult</a></td>
<td>

Update data layer config.

</td>
</tr>
<tr>
<td colspan="2" align="right" valign="top">input</td>
<td valign="top"><a href="#updatedatalayerinput">UpdateDataLayerInput</a>!</td>
<td></td>
</tr>
</tbody>
</table>

#### DataSolutionsSettings

Data solutions settings.

<table>
<thead>
<tr>
<th align="left">Field</th>
<th align="right">Argument</th>
<th align="left">Type</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td colspan="2" valign="top"><strong>isDataLayerEnabled</strong></td>
<td valign="top"><a href="#boolean">Boolean</a>!</td>
<td>

> Indicates if a data layer(bodl) object is enabled for all storefronts.

Reviewer 2: ```suggestion
Indicates enabled data layer (bodl) objects for all storefronts.
```

</tr>
</thead>
<tbody>
<tr>
<td colspan="2" valign="top"><strong>dataSolutions</strong></td>
<td valign="top"><a href="#datasolutionssettings">DataSolutionsSettings</a>!</td>
<td>

Data solutions.

</td>
</tr>
<tr>
<td colspan="2" valign="top"><strong>storeName</strong></td>
<td valign="top"><a href="#string">String</a>!</td>
<td>

Store name.

</td>
</tr>
</tbody>
</table>

#### StoreSettingsMutations

Store Settings mutations

<table>
<thead>
<tr>
<th align="left">Field</th>
<th align="right">Argument</th>
<th align="left">Type</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td colspan="2" valign="top"><strong>dataSolutions</strong></td>
<td valign="top"><a href="#datasolutionsmutations">DataSolutionsMutations</a>!</td>
<td>

Data solutions mutations.

</td>
</tr>
</tbody>
</table>

#### UpdateDataLayerResult

Result of mutation.

<table>
<thead>
<tr>
<th align="left">Field</th>
<th align="right">Argument</th>
<th align="left">Type</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td colspan="2" valign="top"><strong>dataSolutions</strong></td>
<td valign="top"><a href="#datasolutionssettings">DataSolutionsSettings</a></td>
<td>

> Data solutions that is updated as a result of mutation.

Reviewer 2: ```suggestion
Data solutions that are updated as a result of mutation.
```

</td>
</tr>
</tbody>
</table>

### Inputs

#### UpdateDataLayerInput

<table>
<thead>
<tr>
<th colspan="2" align="left">Field</th>
<th align="left">Type</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td colspan="2" valign="top"><strong>isDataLayerEnabled</strong></td>
<td valign="top"><a href="#boolean">Boolean</a>!</td>
<td>

> is data layer enabled

Reviewer 2: ```suggestion
Is data layer enabled?
```

</td>
</tr>
</tbody>
</table>

### Scalars

#### Boolean

The `Boolean` scalar type represents `true` or `false`.

#### String

> The `String` scalar type represents textual data, represented as UTF-8 character sequences. The String type is most often used by GraphQL to represent free-form human-readable text.

Reviewer 2: ```suggestion
The `String` scalar type represents textual data as UTF-8 character sequences. GraphQL most often uses the String type to represent free-form human-readable text.
```

# Data Layer API Reference

The [Data Layer API](/api-docs/store-management/data-layer-api) is a GraphQL management API that allows you to enable and disable the [Big Open Data Layer](/api-docs/analytics/bodl-for-storefronts) for a store.

**Request URL:** 

`https://api.bigcommerce.com/stores/{store_hash}/graphql`

**Path Parameters:**

| Name | Type | Description | Required |
| :--- | :--- | :--- | :--- |
| `store_hash` | string | Unique ID for a store | Yes |

## Authentication

### OAuth scopes

| UI Name | Permission | Parameter |
| :--- | :--- | :--- |
| Information & Settings | modify | `store_v2_information` |

> | Information & Settings | read-only | `store_v2_information_read_only` |

Reviewer: I have not checked how the read-only scope works for this endpoint
Have you tested this case?

Author: I just checked it, and it turns out read-only doesn't work for the queries, so I decided to remove it. Thanks for pointing it out 👍
